### PR TITLE
Fix lock, add fifo strategy

### PR DESCRIPTION
**Pull request by Dirk007**

### CODE REVIEW

File: pkg/fifo/fifo.go [hunk 1]
```diff
@@ -1,101 +1,97 @@
 package fifo
 
 import (
-	"github.com/Dirk007/simpleFifo/pkg/fifo/internal/item"
+	"github.com/Dirk007/simpleFifo/pkg/fifo/implementations"
```
Comment (TiDe-bit) [inline pkg/fifo/fifo.go:4]: a fifo gets called like `fifo.NewFifo().WithImplementation(implementations.NewDoubleLinkedFiFo())` not shure if this isn't a bit wordy

**Review by TiDe-bit — APPROVED** [review]

File: pkg/fifo/fifo.go (hunk 1 above)
Comment (Dirk007) [reply pkg/fifo/fifo.go:4]: bessere Idee? Ich bin da leidenschaftslos.
